src/FacilitatorOptionParser.ts

```diff
@@ -0,0 +1,179 @@
+import { Config, FacilitatorConfig } from './Config';
+import MosaicConfig from './MosaicConfig';
+import { FacilitatorStartException } from './Exception';
+
+/**
+ * It parses and evaluates input params of facilitator start command to get config object.
+ */
+export default class FacilitatorOptionParser {
+  public originChain: string;
+
+  public auxChain: string;
+
+  public mosaicConfigPath: string;
+
+  public facilitatorConfigPath: string;
+
+  /**
+   * Constructor
+   * @param originChain Name of the origin chain.
+   * @param auxChain Identifier of the aux chain.
+   * @param mosaicConfigPath Path to mosaic config file.
+   * @param facilitatorConfigPath Path to facilitator config path.
+   */
+  public constructor(
+    originChain: string,
+    auxChain: string,
+    mosaicConfigPath: string,
+    facilitatorConfigPath: string,
+  ) {
+    this.originChain = originChain === undefined ? '' : originChain.trim();
+    this.auxChain = auxChain === undefined ? '' : auxChain.trim();
+    this.mosaicConfigPath = mosaicConfigPath;
+    this.facilitatorConfigPath = facilitatorConfigPath;
+  }
+
+  /**
+   * It returns Config object encapsulating mosaic and facilitator configs.
+   */
+  public getConfig(): Config {
+    let configObj = {} as any as Config;
+    let onlyFacilitator = false;
+    if (this.isOnlyFacilitatorConfigPath()) {
+      onlyFacilitator = true;
+      configObj = this.handleFacilitatorConfigOption();
+    }
+
+    if (onlyFacilitator === false) {
+      this.isOriginAuxChainDefined();
+      configObj = this.handleOriginAuxChainOption();
+    }
+
+    return configObj;
+  }
+
+  /**
+   * This method returns Config object when origin chain and aux chain is defined.
+   * @returns Config object encapsulating facilitator and mosaic configs.
+   */
+  private handleOriginAuxChainOption(): Config {
```

Comment:
I believe, we can simplify the implementation. Basically, we should read facilitator and mosaic objects either from default location or from the path if the path is specified. Afterwards, we should have one single place where we create and return Config object.